### DISCUSSION TITLE
Sandboxing an entire user

### DISCUSSION THREAD

**Discussion by jbmorris289** [sandboxie-plus/Sandboxie#1759] [General]
Hi. I hope this is the right category.

I'm currently trying to create a "sandboxed" user, where any program or file that is launched is sandboxed. I have ran into a few problems and am confused on how to continue. (I'm on Windows 10.)

Firstly, whenever I open the box settings, Sandboxie asks if I want to enable it (as it shows as disabled). Pressing yes resets the settings that were set previously.
I was able to sandbox all the browsers using the templates. With Edge, however, SBIE2101 messages keep appearing.

> SBIE2101 Object name not found: Unnamed object, error OpenProcess (C0000022) access=00001410 initialized=1

I also have no idea what folders to set, and setting the profile folder broke a couple of things. 

Is something like this possible with Sandboxie Plus?

**Comment by isaak654** [comment]
- Make sure to not have `HideOtherBoxes=y` applied in the sandbox (that is on your C:\Windows\Sandboxie.ini file) if your Plus version is between 1.0.10 and 1.0.15:
#1743
 #1666
- There could be a few browser extensions or programs that could cause other messages. About extensions, you may even need to disable them from the sandboxed browsers. However, the software compatibility could still be improved with more contributors willing to share their efforts in opening or closing the accesses to the incompatible program or extension.

The lead developer recently shared a screenshot about how to allow or deny your resource access:

<img width="75%" src="https://user-images.githubusercontent.com/12372772/161425270-3fdc8304-d053-4fdb-9476-20237311e0fa.png">

### A few notes
* the step 2 provided in the picture (which is the Monitor mode) is not available in all versions before v1.0.16
* instead of step 3, you can save the log in a new file with a left-click to the floppy icon on the right
* optionally, see also [TraceLog](https://sandboxie-plus.github.io/sandboxie-docs/Content/TraceLog.html)

**Reply by jbmorris289** [reply]
Hi.

So I removed the `HideOtherBoxes` parameter for the sandbox in the ini file. 
About settings: because the sandbox appears as disabled for some reason (even though I didn't disable it), I get this message:
![image](https://user-images.githubusercontent.com/22121073/163134273-c00febe6-8932-4ea2-82a6-766988144f15.png)

...and if I were to press `Yes`, all of the settings that were set previously are lost and reset to default. I'm wondering if it has something to do with being only enabled to the specific user I've set?

**Reply by isaak654** [reply]
> I'm wondering if it has something to do with being only enabled to the specific user I've set?

Not necessarily, that dialog can also appear if you double-click a sandbox that was previously set as hidden by switching manually the setting `Enabled=y` to `Enabled=n` in the ini file.

For more information, see the [Enabled](https://sandboxie-plus.github.io/sandboxie-docs/Content/Enabled.html) page.
Although most of the documentation is for Sandboxie Classic, this setting is still relevant on Plus.

> if I were to press Yes, all of the settings that were set previously are lost and reset to default.

Not for me on Plus 1.0.15, in my case it just changed `Enabled=n` to `Enabled=y` (but I didn't set any user before).

EDIT: Tracked as #1792

**Reply by jbmorris289** [reply]
> Not necessarily, that dialog can also appear if you double-click a sandbox that was previously set as hidden by switching manually the setting Enabled=y to Enabled=n in the ini file.

That's something I forgot to mention. It does show as enabled in the ini file, along with the user I've set it to. 
Reading the linked page, that explains why the sandbox is hidden. In testing though, the sandbox does work.

> Not for me on Plus 1.0.15, in my case it just changed Enabled=n to Enabled=y (but I didn't set any user before).

That's weird... it looks like I am on that same version:
![image](https://user-images.githubusercontent.com/22121073/163255828-f51ee17c-2d4c-4074-80ab-98828d3a8885.png)

**Comment by DavidXanatos** [comment]
Please clarify was the box config reset or only the user removed?

**Reply by jbmorris289** [reply]
Both - I have found myself having to set the user each time I'm looking to edit the config through the UI. Editing through the INI file does not have this issue, (if that helps to mention).
